networkx/utils/configs.py

```diff
@@ -258,17 +264,29 @@ class NetworkXConfig(Config):
     - ``NETWORKX_BACKEND_PRIORITY``: set ``backend_priority`` from comma-separated names.
     - ``NETWORKX_CACHE_CONVERTED_GRAPHS``: set ``cache_converted_graphs`` to True if nonempty.
 
+    ``backend`` and ``backend_priority`` configurations are similar in that they can
+    both be used to run an algorithm with a backend (converting inputs if necessary),
+    but they have important differences. ``backend_priority`` is "best effort" and will only
+    use one of the specified backends if it is able to run the algorithm. This is a
+    safer option that behaves well--it doesn't raise--when backends are incomplete.
+    ``backend`` configuration is "strict" and directs all dispatchable calls to use
+    the specified backend. It will raise if the backend does not implement a function.
+
     This is a global configuration. Use with caution when using from multiple threads.
     """
 
+    backend: str | None
```

Comment:
This `backend` config is not needed for this PR, right?
I think this is a part of the PR that you made which adds `backend="networkx"`. 
Or am I missing something and this is needed here?

if we had used `backend_priority = ["nx_loopback"]`, then I think it would have been sufficient for this PR.


Also, I think `backends` and `backend` can cause confusions, and a user can confuse `backends` with `backend_priority`. So, I think it might be nice to revisit my [this comment](https://github.com/networkx/networkx/pull/7485#issuecomment-2161330675) in PR#7485 which proposed to combine the behaviour of `backend` and `backend_priority`. And I know you addressed it by saying that this would restrict how the both(`backend` and `backend_priority`) get developed in the future(but, I'm not completely sure how `backend` can grow further...). So, I am just trying to convey the point that we should name and define the functionalities of these parameters a little better.

Reply:
Good catch! I'll remove this.

This branch was originally based off of #7485, so this must have accidentally slipped through. I agree with not having `nx.config.backend` in this PR, because it needs more discussion. Perhaps we can discuss options at an nx dispatching meeting soon.

Reply:
Okay, I removed `nx.config.backend` from this PR :+1: